src/journal/managed-journal-file.c

```diff
@@ -73,9 +74,15 @@ static int managed_journal_file_entry_array_punch_hole(JournalFile *f, uint64_t
                 return 0;
 
         if (p == le64toh(f->header->tail_object_offset) && !f->seal) {
+                ssize_t n;
+
                 o.object.size = htole64(offset - p);
-                if (pwrite(f->fd, &o, sizeof(EntryArrayObject), p) < 0)
+
+                n = pwrite(f->fd, &o, sizeof(EntryArrayObject), p);
+                if (n < 0)
                         return log_debug_errno(errno, "Failed to modify entry array object size: %m");
+                if ((size_t) n != sizeof(EntryArrayObject))
+                        return log_debug_errno(SYNTHETIC_ERRNO(EIO), "Short pwrite() while modifying entry array object size.");
```

Comment:
Technically, even if we do a short write, we might still have managed to write the object size which is the only thing we really care about. Maybe the call to pwrite() should be changed to only try to write up until and including the size of the object (so replace `sizeof(EntryArrayObject)` with `offsetof(Object, object.field) + sizeof(le64_t)`)? Kind of a nit though so setting the green label